src/sentry/middleware/ratelimit.py

```diff
@@ -23,20 +23,29 @@
 )
 
 
-class RatelimitMiddleware(MiddlewareMixin):
-    """Middleware that applies a rate limit to every endpoint."""
+class RatelimitMiddleware:
+    """Middleware that applies a rate limit to every endpoint.
+    See: https://docs.djangoproject.com/en/4.0/topics/http/middleware/#writing-your-own-middleware
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request: Request) -> Response:
+        # First, check if the endpoint call will violate.
+        data = {}
 
-    def process_view(self, request: Request, view_func, view_args, view_kwargs) -> Response | None:
-        """Check if the endpoint call will violate."""
         try:
-            # TODO: put these fields into their own object
-            request.will_be_rate_limited = False
             request.rate_limit_category = None
```

Comment:
I pretty much only kept this in the `request` object so that we could still do `test_rate_limit_category`, otherwise I didn't know how to test it since we don't store that anywhere else.